src/Segment.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import Constants from 'expo-constants';
+import {
+  View,
+  Animated,
+  ScrollView,
+  SafeAreaView,
+  TouchableNativeFeedback,
+  Dimensions,
+  Text,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import GalioTheme, { withGalio } from './theme';
+
+const {
+  Provider: TilesDimensionPropProvider,
+  Consumer: TilesDimensionPropConsumer,
+} = React.createContext([]);
```

Comment:
I don't think using context is right for this component. I'd actually prefer using `state` memorizing which of the Segment's tab is the active one. Maybe we could try having a `dataArray` prop which will accept an array of components and then render all of those components with they're own onPress function which the user could also edit.

For example let's say: `<Segment dataArray={data} onPress={(option) => {/*do something here*/}}/>` and the data array would look something like this: `data = [<Text>Option 1</Text>, <Text>Option 2</Text>];` 

What do you think?